### PR TITLE
Implements uBO style polyfills for requests redirects

### LINKED ISSUE #27: Handle uBO style polyfills for blocked requests
uBO defines polyfills, or alternate content bodies that should be returned for resources from a URL that matches a pattern.  The bodies of these responses, and identifiers for them, are given here:

https://github.com/uBlockOrigin/uAssets/blob/master/filters/resources.txt

Rules then reference these polyfils with the `redirect=<identifier>` option on rules.

It would be good if adblock-rust both could 
1) understand the redirect option in rules
2) parse polyfil definitions like https://github.com/uBlockOrigin/uAssets/blob/master/filters/resources.txt 
3) and return the alternate message body when appropriate

### PR DESCRIPTION
**Pull request by AndriusA**
returns request content to use for requests matched on rules with `redirect` option

Closes #27

### CODE REVIEW

**Comment by AndriusA** [comment]
@snyderp if you review `tests/ublock-coverage.rs` and the tests in `src/resources.rs` it should be clear what gets handled - please check it is all as intended.

Also added a test that checks parsing extracts all 110 resources for the snapshot of `resources.txt` included in the repo

File: src/blocker.rs [hunk 1]
```diff
@@ -95,11 +100,28 @@ impl Blocker {
...
+                    if let Some(resource) = blocker_redirects.get_resource(redirect) {
+                        let mut data_url: String;
+                        if resource.content_type.contains(';') {
+                            data_url = format!("data:{},{}", resource.content_type, resource.data);
+                        } else {
+                            data_url = format!("data:{};base64,{}", resource.content_type, base64::encode(&resource.data));
+                        }
+                        Some(data_url.trim().to_owned())
+                    } else {
+                        // TOOD: handle error - throw?
```
Comment (snyderp) [inline src/blocker.rs]: Could this be logged somehow, even if its just mirroring the "didn't understand filter: X" stuff the current lib does?  Would be a nice, noisy reminder if there is some new filter format we don't support, something like that

**Comment by snyderp** [comment]
Also, is this exposed through the node API?

**Comment by AndriusA** [comment]
@snyderp should we merge and close? It is backwards-compatible, so don't see any harm in that, FFI and browser can be wired up in due course

**Comment by snyderp** [comment]
Yep, looks good to me, though it looks like there are some conflicts now.  But I think the change is terrific!

**Comment by AndriusA** [comment]
Merged master to resolve conflicts

Also to ensure deserialization is backwards compatible have to ignore deserializing of redirect resources for now - included the required changes, tagging @bbondy for review

**Comment by AndriusA** [comment]
Switching to [MessagePack](https://msgpack.org) for serialization to have a cleaner way of adding new fields to blocker structures over time - this works as intended with `#serde(default)` field property by using default value for the field if it isn't found when deserializing.

**Pull request merged**